zonemaster/zonemaster-gui
# Improve result UI

## Purpose

Improve overall result UI

## Context

* Addresses an horizontal overflow issue on mobile screens
* Fix navigation bar animation
* Addresses #115

## Changes

* Dependencies upgrade (angular, typescript, ngx-translate, ng-bootstrap)
* Make the module header sticky
* Refactor result component HTML (tables do not support sticky element very well)
* Hide the module column on mobile device (actually I wonder if this column is still useful on wide screen as well now that the module header is sticky, any thoughts ?)
* Fix the dimension of the github thingy
* Fix the navigation bar animation
* Fix "back to top" button
* Set a fixed height (80px) for the nav bar on wide screens
* Make the generation of the HTML export (almost) independent from the component rendering (so that changing the component template don't affect the export).
* Attempt to fix blurry logo on Chrome
* Fix the filters / action buttons alignment on narrow devices (#115)
* Remove the "ungroup" feature
* Remove the header of the result table


## How to test this PR

* Make a test
* Expend some modules in the result, enough to scroll
* Check that the module name is sticky, just below the nav bar on both mobile and desktop screens
* Clicking the sticky header should collapse the module but the header should not move because of the scroll offset when collapsing the module
* The module column should not be visible on narrow devices
* There should be no horizontal overflow on any devices

<img src="https://user-images.githubusercontent.com/19394895/130472299-a785bf66-e2db-48c5-b65d-259ed3e51621.jpg" width="50%">

<img src="https://user-images.githubusercontent.com/19394895/131349511-0de5d0c4-b469-497b-8c47-d3f0c84c9289.jpg" width="50%">
